api/v1alpha1/drpolicy_types.go

```diff
@@ -64,11 +78,25 @@ type DRPolicySpec struct {
 	DRClusterSet []ManagedCluster `json:"drClusterSet"`
 }
 
+// +kubebuilder:validation:Enum=Unfenced;Fenced
+type FenceStatus string
+
+const (
+	ClusterFenced   = FenceStatus("Fenced")
+	ClusterUnfenced = FenceStatus("Unfenced")
+)
+
+type ClusterStatus struct {
+	Name   string      `json:"string,omitempty"`
+	Status FenceStatus `json:"status,omitempty"`
+}
+
 // DRPolicyStatus defines the observed state of DRPolicy
 // INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 // Important: Run "make" to regenerate code after modifying this file
 type DRPolicyStatus struct {
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	Conditions []metav1.Condition       `json:"conditions,omitempty"`
+	DRClusters map[string]ClusterStatus `json:"drClusters,omitempty"`
```

Comment:
I lean towards conditions, as for example in this case say gen 1 of the spec was fencing e1, and then gen 2 removed the fence, and gen 3 again fenced e1, if status was not updated post gen 2, then a user/controller looking at status when spec changed to gen 3 would assume e1 is already fenced.

Conditions add the `observedGeneration` that helps with the above, and also we can represent one condition, say stating (very broadly) `Reconciled`, which would be T|F for the generation.

The Condition and the spec is looked at to see what is met, i.e for example "gen 3 is reconciled to it's spec, which states e1 is fenced"

(can discuss and change later)

Reply:
Agree. IIUC a good example would be how VRG manages its status via per PVC conditions and VRG having its own conditions updated based on per PVC conditions. Is this assumption correct?

I think we will get there (i.e. conditions in general, more precisely the above example if that looks OK). For now, DRPolicy (or any ramen component) has the functionality to do fencing of a managed cluster. Thus, for now only manuallyFenced capability is available with DRPolicy and that is why just an array for the managedCluster status (mainly for fencing).

DRPolicy currently uses only one ConditoinType (for saying if DRPolicy is validated or not). So, we are thinking to handle the status via conditions properly once we start adding ramen initiated fencing capability (for now DRPolicy). A comment has been added [1] to address this later.

[1] https://github.com/RamenDR/ramen/pull/370/files#diff-92d35b53448e8c5810485f47933e5fc16fcfcc32b5cbe3ddbc5325961b2bd7a8R104